src/core/control/Control.cpp

```diff
@@ -1979,7 +1992,35 @@ void Control::toolLineStyleChanged() {
     const LineStyle& lineStyle = toolHandler->getTool(TOOL_PEN).getLineStyle();
     string style = StrokeStyle::formatStyle(lineStyle);
 
-    if (style == "dash") {
+    bool isLineStyleSameForAll = true;
+	
+    if (win) {
+        EditSelection* sel = win->getXournal()->getSelection();
+        if (sel) {
+			bool isFirstStrokeElement = true;
+			string previous_style = "none";
+			
+			for (Element* e: sel->getElements()) {
+				if (e->getType() == ELEMENT_STROKE) {
+					auto* s = dynamic_cast<Stroke*>(e);
+					style = StrokeStyle::formatStyle(s->getLineStyle());
```

Comment:
You may wanna filter with `s->getToolType() == STROKE_TOOL_PEN` as well here, to exclude highlighter or whiteout strokes.

Reply:
Thanks, resolved by 67e5e9dea630e9ed617d84e44be6078f89b6ecee and the later refactorings.